### PR TITLE
Changes for refactoring Org creation and introduced user context

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/7393

*Description of changes:*

validators introduced. changes in the org service.
user_Context reused from pay-api

Not Done:
couldnt make update and create of org together. Leads to more code/gets complicated.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/tests/conftest.py [hunk 1]
```diff
@@ -292,3 +292,51 @@ def get_nr(business_identifier, bearer_token):
...
+@pytest.fixture()
+def bceid_user_mock(monkeypatch):
+    """Mock user_context."""
+
+    def token_info():  # pylint: disable=unused-argument; mocks of library methods
+        return {
+            'username': 'CP1234567 user',
+            'realm_access': {
+                'roles': [
+                    'staff',
```
Comment (mengdong19) [inline auth-api/tests/conftest.py:330]: bceid user should not have staff role.

File: auth-api/src/auth_api/utils/user_context.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        return self._user_name.upper() if self._user_name else None
+
+    @property
+    def first_name(self) -> str:
+        """Return the user_name."""
+        return self._first_name
+
+    @property
+    def user_id(self) -> str:
+        """Return the user_name."""
```
Comment (mengdong19) [inline auth-api/src/auth_api/utils/user_context.py:65]: """Return the user_id."""

File: auth-api/tests/conftest.py [hunk 3]
```diff
@@ -292,3 +292,51 @@ def get_nr(business_identifier, bearer_token):
...
+                ]
+            }
+        }
+
+    def mock_auth():  # pylint: disable=unused-argument; mocks of library methods
+        return 'test'
+
+    monkeypatch.setattr('auth_api.utils.user_context._get_token', mock_auth)
+    monkeypatch.setattr('auth_api.utils.user_context._get_token_info', token_info)
+    monkeypatch.setattr('auth_api.utils.user_context._get_token_info', token_info)
```
Comment (mengdong19) [inline auth-api/tests/conftest.py:318]: duplicated lines?

File: auth-api/tests/conftest.py [hunk 4]
```diff
@@ -292,3 +292,51 @@ def get_nr(business_identifier, bearer_token):
...
```
Comment (mengdong19) [inline auth-api/tests/conftest.py:342]: duplicate lines.

File: auth-api/tests/conftest.py (hunk 1 above)
Comment (saravanpa-aot) [reply auth-api/tests/conftest.py]: changed.thanks

File: auth-api/src/auth_api/utils/user_context.py (hunk 2 above)
Comment (saravanpa-aot) [reply auth-api/src/auth_api/utils/user_context.py]: changed.thanks

File: auth-api/tests/conftest.py (hunk 3 above)
Comment (saravanpa-aot) [reply auth-api/tests/conftest.py]: done

File: auth-api/tests/conftest.py (hunk 4 above)
Comment (saravanpa-aot) [reply auth-api/tests/conftest.py]: done

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**